lib/workers/repository/reconfigure/index.ts

```diff
@@ -0,0 +1,165 @@
+import is from '@sindresorhus/is';
+import JSON5 from 'json5';
+import type { RenovateConfig } from '../../../config/types';
+import { validateConfig } from '../../../config/validation';
+import { logger } from '../../../logger';
+import { platform } from '../../../modules/platform';
+import { ensureComment } from '../../../modules/platform/comment';
+import { scm } from '../../../modules/platform/scm';
+import { getCache } from '../../../util/cache/repository';
+import { readLocalFile } from '../../../util/fs';
+import { getBranchCommit } from '../../../util/git';
+import { regEx } from '../../../util/regex';
+import { detectConfigFile } from '../init/merge';
+import {
+  deleteReconfigureBranchCache,
+  setReconfigureBranchCache,
+} from './reconfigure-cache';
+
+export function getReconfigureBranchName(prefix: string): string {
+  return `${prefix}reconfigure`;
+}
+export async function validateReconfigureBranch(
+  config: RenovateConfig
+): Promise<void> {
+  logger.debug('validateReconfigureBranch()');
+  const context = `renovate/config-validation`;
+
+  const branchName = getReconfigureBranchName(config.branchPrefix!);
+  const branchExists = await scm.branchExists(branchName);
+
+  // this is something the user initiates, so skip if no branch exists
+  if (!branchExists) {
+    logger.debug('No reconfigure branch found');
+    deleteReconfigureBranchCache(); // in order to remove cache when the branch has been deleted
+    return;
+  }
+
+  // look for config file
+  // 1. check reconfigure branch cache and use the configFileName if it exists
+  // 2. checkout reconfigure branch and look for the config file, don't assume default configFileName
+  const branchSha = getBranchCommit(branchName)!;
+  const cache = getCache();
+  let configFileName: string | null = null;
+  const reconfigureCache = cache.reconfigureBranchCache;
+  const validationStatus = await platform.getBranchStatusCheck(
+    branchName,
+    'renovate/config-validation'
+  );
+
+  // only use valid cached information
+  if (
+    reconfigureCache?.reconfigureBranchSha === branchSha ||
+    validationStatus !== null
+  ) {
+    logger.debug('Skipping validation check');
+    return;
+  }
```

Comment:
```suggestion
  // only use valid cached information
  if (reconfigureCache?.reconfigureBranchSha === branchSha) {
    logger.debug('Skipping validation check as branch sha is unchanged');
    return;
  }
  
  const validationStatus = await platform.getBranchStatusCheck(
    branchName,
    'renovate/config-validation'
  );
```

Do it like this to avoid an API call if the branch SHA is unchanged, if you don't care about the result